### ISSUE TITLE
Bad arity checking for non-linear rule

### ISSUE THREAD

**Issue by rafoo** [Deducteam/Dedukti#57]
In the following file, Dedukti fails to ensure that the right-hand-side uses the variable `f` with arity at least one. The rule is accepted but not always used as shown by the assertion.

```
#NAME bug.

type : Type.
def term : type -> Type.

arr : type -> type -> type.
[A,B] term (arr A B) --> term A -> term B.

E : A : type -> term A -> type.
r : A : type -> a : term A -> term (E A a).

A : type.
B : type.

def e : f : (term A -> term B) ->
        (x : term A -> term (E B (f x))) ->
        term (E (arr A B) f).
[f] e f (x => r B (f x)) --> r (arr A B) f.

#ASSERT (f : term (arr A B) => e f (x : term A => r B (f x))) ==
        (f : term (arr A B) => r (arr A B) f).
```

Error message: `Fatal error: exception Failure("Assertion failed.")`

**Comment by francoisthire** [comment]
My guess is that because of Miller's pattern, the matching problem for the second occurrence of `f` is eta-expanded hence the conversion failed.

**Comment by rafoo** [comment]
My point is that I should not be allowed to write the rewrite rule without eta-expanding `f` in the right-hand side.

**Comment by francoisthire** [comment]
The following example works however:
```
type : Type.
def term : type -> Type.

arr : type -> type -> type.
[A,B] term (arr A B) --> term A -> term B.

E : A : type -> term A -> type.
r : A : type -> a : term A -> term (E A a).

A : type.
B : type.

def e : f : (term A -> term B) ->
        (x : term A -> term (E B (f x))) ->
        term (E (arr A B) f).
[f] e f (x => r B (f x)) --> r (arr A B) f.

def left := (f : term (arr A B) => e (x : term A => f x) (x : term A => r B (f x))).
def right := (f : term (arr A B) => r (arr A B) (x : term A => f x)).

#ASSERT left == right.
```

We also need to write `(x => f x)` for the first occurence of `f` in the rule. Note that for the conversion test works, Dedukti needs an eta-expansion for the term `right`.

**Comment by francoisthire** [comment]
Fixed in #60
